app/models/course/survey/answer_option.rb

```diff
@@ -3,4 +3,13 @@ class Course::Survey::AnswerOption < ActiveRecord::Base
   belongs_to :answer, inverse_of: :options
   belongs_to :question_option, class_name: Course::Survey::QuestionOption.name,
                                inverse_of: :answer_options
+
+  validate :question_option_belongs_to_question
+
+  private
+
+  def question_option_belongs_to_question
+    errors.add(:question_option_id, :invalid) \
+      unless answer.question.option_ids.include?(question_option_id)
```

Comment:
Have added a commit to validate the input params instead of the answer option itself. Saved some queries through memoisation. If okay, I'll squash.

However, now answer options are created/destroyed when users selects/unselects options. This is not done in bulk, so there are still ~N queries.